test/conformance/tests/sriov_operator.go

```diff
@@ -139,61 +140,28 @@ var _ = Describe("[sriov] operator", func() {
 		resourceName := "testresource"
 		var node string
 		var sriovDevice *sriovv1.InterfaceExt
+		var discoveryModeSkip bool
 
 		execute.BeforeAll(func() {
-			node = sriovInfos.Nodes[0]
 			var err error
-			sriovDevice, err = sriovInfos.FindOneSriovDevice(node)
-			Expect(err).ToNot(HaveOccurred())
 
-			// For the context of tests is better to use a Mellanox card
-			// as they support all the virtual function flags
-			// if we don't find a Mellanox card we fall back to any sriov
-			// capability interface and skip the rate limit test.
-			intf, err := sriovInfos.FindOneMellanoxSriovDevice(node)
-			if err != nil {
-				intf, err = sriovInfos.FindOneSriovDevice(node)
-				Expect(err).ToNot(HaveOccurred())
-			}
+			if discovery.Enabled() {
+				node, resourceName, numVfs, err = discovery.DiscoveredResources(clients, sriovInfos.Nodes, operatorNamespace)
+				discoveryModeSkip = node == "" || resourceName == "" || numVfs < 5 || err != nil
 
-			config := &sriovv1.SriovNetworkNodePolicy{
-				ObjectMeta: metav1.ObjectMeta{
-					GenerateName: "test-policy",
-					Namespace:    operatorNamespace,
-				},
-
-				Spec: sriovv1.SriovNetworkNodePolicySpec{
-					NodeSelector: map[string]string{
-						"kubernetes.io/hostname": node,
-					},
-					NumVfs:       numVfs,
-					ResourceName: resourceName,
-					Priority:     99,
-					NicSelector: sriovv1.SriovNetworkNicSelector{
-						PfNames: []string{intf.Name},
-					},
-					DeviceType: "netdevice",
-				},
+			} else {
+				node = sriovInfos.Nodes[0]
+				createSriovNetworkNodePolicy(node, sriovInfos, numVfs, resourceName)
 			}
-
-			err = clients.Create(context.Background(), config)
+			sriovDevice, err = sriovInfos.FindOneSriovDevice(node)
```

Comment:
This might not be enough, you will probably care about finding one of the "best resource"

Reply:
Done,
Later we should also make sure that it picks one that the policy can actually use (for example by considering the policy nic selector), but this is outside the scope of this PR